### ISSUE TITLE
What is the accuracy obtained with this Keras implementation?

### ISSUE THREAD

**Issue by vwrobel** [fferroni/DEC-Keras#2]
Hello @fferroni. Thank you for sharing this implementation.

With the MXNET model, I can get a 82% clustering accuracy on MNIST but get stuck around 60% with this Keras model out of the box?

Could you tell be if you manage to reach a higher accuracy? Thanks.

**Comment by fferroni** [comment]
Yes I noticed that too. I suspect it has something to do with the autoencoder pre-training, because the clustering accuracy right after pre-training is already very good in the paper, but not so in this case. I tried to keep things "like-for-like" as in the MXNET code and I cannot see any glaring difference...

**Comment by vwrobel** [comment]
I have arrived to the same conclusion regarding the auto-encoder pre-training. Cheers.

**Comment by nadavbar** [comment]
@fferroni  and @vwrobel - I have forked this (great) implementation and changed the pre-training to be like the paper - e.g. layer-wise pre-training for the encoder and removed the dropout for the second phase of the training like they do in the paper. It's a bit hard to see this in the MXNET code since it already part of their [autoencoder implementation](https://github.com/dmlc/mxnet/blob/master/example/autoencoder/autoencoder.py#L134).
I get 84% precent on MNIST with this like the paper. The code is available here: https://github.com/nadavbar/DEC-Keras
@fferroni  - If you are interested I can do a pull request to this repo.

**Comment by fferroni** [comment]
Nice! Great job. Yes, I'd really appreciate if you could do a pull request.

**Comment by nadavbar** [comment]
@fferroni  - Thanks! I've opened up a PR
